expressServer/index.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const cors = require('cors');
+const dotenv = require('dotenv');
+const { uniqueNamesGenerator, adjectives, colors, animals } = require('unique-names-generator');
+
+dotenv.config();
+
+const app = express();
+const port = process.env.PORT || 5000;
+
+app.use(cors());
+app.use(express.json());
+
+mongoose.connect(process.env.MONGODB_URI)
+  .then(() => console.log('Connected to MongoDB'))
+  .catch(err => console.error('Could not connect to MongoDB', err));
+
+const userSchema = new mongoose.Schema({
+  username: { type: String, required: true },
+  password: { type: String, required: true },
+});
+
+const User = mongoose.model('User', userSchema);
+
+const generateRandomName = () => uniqueNamesGenerator({ dictionaries: [adjectives, colors, animals] });
+
+app.post('/api/login', async (req, res) => {
+  const { username, password } = req.body;
+
+  try {
+    const user = await User.findOne({ username });
+
+    if (!user || user.password !== password) {
+      return res.status(401).json({ message: 'Incorrect username or password' });
+    }
+
+    res.status(200).json({ message: 'Login successful', username });
+  } catch (error) {
+    console.error('Error during login:', error);
+    res.status(500).json({ message: 'Internal server error' });
+  }
+});
+
+app.get('/api/snakes', async (_, res) => {
```

Comment:
Do you use this route in the frontend?

Reply:
Not yet, I plan on using it in the future.